pangeo_forge_recipes/transforms.py

```diff
@@ -647,22 +611,37 @@ class StoreToZarr(beam.PTransform, ZarrWriterMixin):
       based on the full dataset (e.g. divide along a certain dimension based on a desired chunk
       size in memory). For more advanced chunking strategies, check
       out https://github.com/jbusecke/dynamic_chunks
-    :param dynamic_chunking_fn_kwargs: Optional keyword arguments for ``dynamic_chunking_fn``.
     :param attrs: Extra group-level attributes to inject into the dataset.
     :param encoding: Dictionary encoding for xarray.to_zarr().
     :param append_dim: Optional name of the dimension to append to.
+
+    Example of using a wrapper function to reduce the arity of a more complex dynamic_chunking_fn:
+
+        Suppose there's a function `calculate_dynamic_chunks` that requires extra parameters: an
+        `xarray.Dataset`, a `target_chunk_size` in bytes, and a `dim_name` along which to chunk.
+        To fit the expected signature for `dynamic_chunking_fn`, we can define a wrapper function
+        that presets `target_chunk_size` and `dim_name`:
+
+        ```python
+        def calculate_dynamic_chunks(ds, target_chunk_size, dim_name) -> Dict[str, int]:
+            ...
+
+        def dynamic_chunking_wrapper(ds: xarray.Dataset) -> Dict[str, int]:
+            target_chunk_size = 1024 * 1024 * 10
+            dim_name = 'time'
+            return calculate_dynamic_chunks(ds, target_chunk_size, dim_name)
+
+        StoreToZarr(..., dynamic_chunking_fn=dynamic_chunking_wrapper, ...)
+        ```
     """
 
-    # TODO: make it so we don't have to explicitly specify combine_dims
-    # Could be inferred from the pattern instead
```

Comment:
I guess we're abandoning this idea? I have had my doubts that it's possible, since it would be a meta-operation over the pipeline. But i've been wrong about what's possible in beam before.

Reply:
The way this reads to me right now, we'd need to pass the same argument to a couple of different transforms (which isn't a bad pattern afaict). To reuse such a value, we'd need to thread the input through as some kind of computed result and I doubt that juice is worth the squeeze

I'm not opposed to keeping the TODO open, if that's desirable. I just couldn't quite imagine how this could be facilitated at the level of recipes